cw_solana/lib/solana_client.dart

```diff
@@ -0,0 +1,406 @@
+      final tokenAmountResult =
+          await _client!.rpcClient.getTokenAccountBalance(programAccount.pubkey);
+
+      final balance = tokenAmountResult.value.uiAmountString;
+
+      final balanceAsDouble = double.tryParse(balance ?? '0.0') ?? 0.0;
+
+      totalBalance += balanceAsDouble;
+    }
+
+    return SolanaBalance(totalBalance);
+  }
+
+  Future<double> getGasForMessage(String message) async {
+    try {
+      final gasPrice = await _client!.rpcClient.getFeeForMessage(message) ?? 0;
+      final fee = gasPrice / lamportsPerSol;
+      return fee;
+    } catch (_) {
+      return 0;
+    }
+  }
+
+  /// Load the Address's transactions into the account
+  Future<List<SolanaTransactionModel>> fetchTransactions(Ed25519HDPublicKey address) async {
+    List<SolanaTransactionModel> transactions = [];
+
+    try {
+      final response = await _client!.rpcClient.getTransactionsList(
+        address,
+        commitment: Commitment.confirmed,
+      );
+
+      for (final tx in response) {
+        if (tx.transaction is ParsedTransaction) {
+          final parsedTx = (tx.transaction as ParsedTransaction);
+          final message = parsedTx.message;
+
+          final fee = (tx.meta?.fee ?? 0) / lamportsPerSol;
+
+          for (final instruction in message.instructions) {
+            if (instruction is ParsedInstruction) {
+              instruction.map(
+                system: (data) {
+                  data.parsed.map(
+                    transfer: (data) {
+                      ParsedSystemTransferInformation transfer = data.info;
+                      bool isOutgoingTx = transfer.source == address.toBase58();
+                      double amount = transfer.lamports.toDouble() / lamportsPerSol;
+
+                      transactions.add(
+                        SolanaTransactionModel(
+                          id: parsedTx.signatures.first,
+                          from: transfer.source,
+                          to: transfer.destination,
+                          amount: amount,
+                          isOutgoingTx: isOutgoingTx,
+                          blockTimeInInt: tx.blockTime!,
+                          fee: fee,
+                          programId: SystemProgram.programId,
+                          tokenSymbol: 'SOL',
+                        ),
+                      );
+                    },
+                    transferChecked: (_) {},
+                    unsupported: (_) {},
+                  );
+                },
+                splToken: (data) {
+                  data.parsed.map(
+                    transfer: (data) {},
+                    transferChecked: (data) {
+                      final mintAddress = tx.meta?.postTokenBalances.first.mint ?? '';
+                      final tokens = DefaultSPLTokens().initialSPLTokens;
+                      final tokenSymbol =
+                          tokens.firstWhere((token) => token.mintAddress == mintAddress).symbol;
```

Comment:
if the user has some tokens that are not in Cake or not enabled in Cake, then this would throw an exception and it will result in all transactions not showing, so just wrap this in a try-catch and add it's default value as SOL if we can't get the token name